jocd/src/br/org/certi/jocd/dapaccess/CmsisDapProtocol.java

```diff
@@ -194,8 +199,29 @@ public byte transferConfigure(byte idleCycles, int waitRetry, int matchRetry)
     return response[1];
   }
 
-  public byte setSWJClock(int freq) {
-    // TODO
-    return 0x00;
+  public int setSWJClock() throws DeviceError, CommandError {
+    return setSWJClock(1000000);
+  }
+
+  public byte setSWJClock(int clock) throws DeviceError, CommandError {
+    byte[] cmd = new byte[2];
```

Comment:
This byte array has 5 elements, so... new byte[5]

Reply:
Fixed